WP-82 - Added additional models

usersblock opened the pull request:
Added sublist,movie,follow and modified user to prevent table pluralizing.

usersblock committed:
Added additional models

Added sublist,movie,follow and modified user to prevent table pluralizing.

PredatorFeesh commented on WatchParty/backend/models/sublist.js:
```diff
@@ -0,0 +1,33 @@
+    name: {
+		type: DataTypes.STRING(32),
+		allowNull: false,
+		unique: 'uniquecombination',
```
I think we also need to add fields https://stackoverflow.com/a/42199137

PredatorFeesh commented on WatchParty/backend/models/movie.js:
```diff
@@ -0,0 +1,29 @@
+    },
+    watchstate: {
+      type: DataTypes.ENUM,
+	  values:['watched','to-watch','not-interested'],
```
Beautiful 🚀

PredatorFeesh reviewed (commented):
One thing I noticed; we don't define `userid` in models. I know the ID is auto generated, but is this the name it's given?

usersblock commented:
I believe it's only called id, but for the sake of differentiating what belongs to what I named it userid.

PredatorFeesh commented on WatchParty/backend/models/sublist.js:
```diff
@@ -0,0 +1,33 @@
+		foreignKey: 'userid',
+		onDelete: 'cascade'
+	});
+	Sublist.belongsTo(models.Movie,{
```
This is a one-to-many relation (we have 1 sublist for many movies https://sequelize.readthedocs.io/en/latest/docs/associations/#one-to-many-associations)

usersblock replied:
Added has many to user and movie

PredatorFeesh commented on WatchParty/backend/models/follow.js:
```diff
@@ -0,0 +1,26 @@
+},{
+    freezeTableName: true
+	});
+	Follow.associate = function(models) {
```
These two are belongs-to-many (you can specify how  here https://sequelize.readthedocs.io/en/latest/docs/associations/#belongs-to-many-associations )

usersblock replied:
Changed both to belongs-to-many.

usersblock committed:
Changed associations

ian-s-mcb reviewed (changes requested):
It would be a good idea to run the backend server before submitting this PR.

Sequelize throws a bunch of errors. I was able to fix them by removing the duplicate table fields. When you specify an associated key in the options object, Sequelize automatically adds those fields to the table, so specifying them again in the schema object causes a duplicate field error. The other error I noticed was that the Sublist model doesn't need an association with the User model because Sublist already has an association with Movie, which provides an association to User. Directly adding an association to User causes a duplicate field error.

usersblock committed:
Changed back associations/Changed Sublist

usersblock commented:
Belongstomany creates a new table to hold relations and  hasmany adds a foreign key to the target. Belongsto is relatively simple as it just connects a foreignkey to another table. To keep each parameter in each model file, I think we should only use Belongsto. As for sublist, I've removed the userid attribute and had unique be applied to name and movie. I've added the change to dbnotes as well.

PredatorFeesh commented:
For BelongsToMany, you can use a `through` table. Whichever way you think it's best done, we'll go with that as long as it can pass the tests and edge cases

usersblock committed:
Fixed unique/validation errors

usersblock committed:
Added unique constraint to sublist

usersblock committed:
Added correct seeds

Removed demo-users
Added userarray,followarray

usersblock committed:
Merge pull request #44 from PredatorFeesh/sandbox

Merging updates into wp-82-additional-models
